ValveResourceFormat/NavMesh/NavMeshFile.cs

```diff
@@ -0,0 +1,208 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using SkiaSharp;
+using ValveResourceFormat.ResourceTypes;
+using ValveResourceFormat.Utils;
+
+namespace ValveResourceFormat.NavMesh
+{
+    public class NavMeshFile
+    {
+        public const uint MAGIC = 0xFEEDFACE;
+
+        public uint Version { get; set; }
+        public uint SubVersion { get; set; }
+
+        public Dictionary<uint, NavMeshArea> Areas { get; private set; }
+        private readonly Dictionary<byte, List<NavMeshArea>> LayerAreas = [];
+        public NavMeshLadder[] Ladders { get; private set; }
+        public int LayerCount { get; private set; }
+        public bool IsAnalyzed { get; private set; }
+
+        public void Read(Stream stream)
+        {
+            using var binaryReader = new BinaryReader(stream, Encoding.UTF8, true);
+            Read(binaryReader);
+        }
+
+        public void Read(BinaryReader binaryReader)
+        {
+            Areas = [];
+            LayerAreas.Clear();
+
+            var magic = binaryReader.ReadUInt32();
+            if (magic != MAGIC)
+            {
+                throw new UnexpectedMagicException($"Unexpected magic, expected {MAGIC:X}", magic, nameof(magic));
+            }
+
+            Version = binaryReader.ReadUInt32();
+            SubVersion = binaryReader.ReadUInt32();
+            IsAnalyzed = binaryReader.ReadUInt32() != 0;
+
+            Vector3[][] polygons = null;
+            if (Version >= 35)
+            {
+                polygons = ReadPolygons(binaryReader);
+                var unk1 = binaryReader.ReadUInt32();
+                var unk2 = binaryReader.ReadUInt32();
+            }
+
+            var areaCount = binaryReader.ReadUInt32();
+            for (var i = 0; i < areaCount; i++)
+            {
+                var area = new NavMeshArea();
+                area.Read(binaryReader, this, polygons);
+                AddArea(area);
+            }
+
+            var ladderCount = binaryReader.ReadUInt32();
+            Ladders = new NavMeshLadder[ladderCount];
+            for (var i = 0; i < ladderCount; i++)
+            {
+                var ladder = new NavMeshLadder();
+                ladder.Read(binaryReader, this);
+                Ladders[i] = ladder;
+            }
+
+            if (Version >= 35)
+            {
+                ReadPostMetadata(binaryReader);
+
+                while (binaryReader.ReadByte() == 0)
+                {
+                    //Reading past 0x00 padding
+                }
+
+                var kv3 = new BinaryKV3();
```

Comment:
I think we should store this, if there's any interesting data.

Reply:
I'd definitely want to, just not sure how yet. I think it's game-specific data, CS2 stores hidingspotdata, spotencounterdata, approachdata, earliestoccupytime for each area.